src/App.js

```diff
@@ -8,56 +8,116 @@ import PetForm from './pages/PetForm'
 import PetProfile from './pages/PetProfile'
 import TaskForm from './pages/TaskForm'
 import Filter from './pages/Filter'
-import { loadFromLocal, saveToLocal } from './services'
-import Pets from './pets.json'
+import { db } from './firebase'
+import { storage } from './firebase'
 
 export default function App() {
-  const [pets, setPets] = useState(loadFromLocal('pets') || Pets)
+  const [pets, setPets] = useState([])
   useEffect(() => {
-    saveToLocal('pets', pets)
-  }, [pets])
+    db.collection('pets').onSnapshot((snapshot) => {
+      const allPets = snapshot.docs.map((doc) => ({
+        id: doc.id,
+        ...doc.data(),
+      }))
+      setPets(allPets)
+    })
+  }, [])
+  const [tasks, setTasks] = useState([])
+  useEffect(() => {
+    db.collection('tasks').onSnapshot((snapshot) => {
+      const allTasks = snapshot.docs.map((doc) => ({
+        id: doc.id,
+        ...doc.data(),
+      }))
+      setTasks(allTasks)
+    })
+  }, [])
+
+  const [previewImage, setPreviewImage] = useState({
+    imageUrl:
+      'https://firebasestorage.googleapis.com/v0/b/pawlog-app.appspot.com/o/images%2Ftaskpaw.png?alt=media&token=8ad10974-93e4-4fd7-ae05-1567d049ad1f',
+    imageName: 'taskpaw.png',
+  })
+
```

Comment:
maybe you can outsource your whole image upload to its own component

Reply:
great idea!